src/datacenters/v1/README.md

```diff
@@ -0,0 +1,216 @@
+
+```yml
+environment:
+  resources:
+    # ...
+  accounts:
+    # ...
+  hooks:
+    # ...
+```
+
+### Environment resources
+
+Like with the root scope, arcctl resources can be declared within the environment
+specification as well. When declared here, one of each resource will be created for
+each environment within the datacenter.
+
+```yml
+environment:
+  resources:
+    namespace:
+      type: namespace
+      # The resources can point to datacenter-level resources
+      account: ${{ accounts.cluster.id }}
+      # The environment name can be injected dynamically
+      name: ${{ environment.name }}
+    nginx-ingress-controller:
+      type: helmChart
+      name: ${{ environment.name }}-ingress-nginx
+      account: ${{ accounts.cluster.id }}
+      # Environment resources can point to each other
+      namespace: ${{ environment.resources.namespace.id }}
+      repository: https://kubernetes.github.io/ingress-nginx
+      chart: ingress-nginx
+```
+
+### Environment accounts
+
+Also as with the root scope, new arcctl accounts can be registered for each enviroment
+that broker access to environment-scoped resources:
+
+```yml
+environment:
+  resources:
+    pg:
+      type: database
+      account: ${{ variables.account }}
+      name: ${{ environment.name }}-pg
+      databaseType: postgres
+      databaseVersion: '13'
+      databaseSize: n/a
+      vpc: n/a
+      region: n/a
+  
+  accounts:
+    pg:
+      name: ${{ environment.name }}-postgres-db
+      provider: postgres
+      credentials:
+        host: ${{ environment.resources.pg.host }}
+        port: ${{ environment.resources.pg.port }}
+        username: ${{ environment.resources.pg.username }}
+        password: ${{ environment.resources.pg.password }}
+        database: architect
+```
+
+### Resource hooks
+
+The last, and possibly most interesting part of the datacenter spec is the ability to define `hooks`.
+Hooks are rules for how application resources should be modified when they land within the datacenter.
+
+These hooks allow operators to assign rules for how ALL resources should behave in the environment and
+are the main feature that enables operators to retain control of the environment without compromising
+developer self-service.
+
+```yaml
+hooks:
+  # Hooks can define "when" clauses that indicate what fields
+  # must match for the hook to apply
+  - when:
+      type: secret
+    account: local
+  - when:
+      type: databaseSchema
+      databaseType: postgres
+    # Hooks can define inline resources that should be created
+    # whenever an application resource matches the hook
+    resources:
+      db:
+        type: database
+        account: ${{ variables.doAccount }}
+        # Inline resources can reference fields from the application 
+        # resource node like .id, .inputs.*, and .outputs.*
+        name: ${{ this.id }}
+        databaseVersion: ${{ this.inputs.databaseVersion }}
```

Comment:
The concept of inputs is not very well defined here. We should explain how these values get mapped from the component into this field so they understand what affects what.